File: start-points/TirePressure/Javascript/alarm.js
'use strict';

const {createSensor} = require('./sensor');

function createAlarm() {
   const lowPressureThreshold = 17;
   const highPressureThreshold = 21;
   const sensor = createSensor();
   let isOn = false;
  
    return {
      check: () => {
         var psiPressureValue = sensor.popNextPressurePsiValue();
         if (psiPressureValue < lowPressureThreshold || highPressureThreshold < psiPressureValue)
         {
           isOn = true;
         }
         
      },
      isOn: () => isOn
    };
}  

module.exports = {createAlarm, check};

File: start-points/TirePressure/CGreenSlice/alarm.c

#include "alarm.h"
#include "sensor.h"

bool ALARM_ON;

void check()
{
    ALARM_ON = false;
    double psiPressureValue = popNextPressurePsiValue_ptr();

    if (psiPressureValue < 17 || 21 < psiPressureValue)
    {
        ALARM_ON = true;
    }
}